### PR TITLE
Island: Fix updating logic for IslandConfigOptions

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do? 

Fixes the logic that updates the Island config options from the package and user-specified config files.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
    > Tested by running blackbox tests for Docker package
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Comment by mssalvatore** [comment]
image

File: monkey/monkey_island/cc/setup/island_config_options.py [hunk 1]
```diff
@@ -20,22 +19,39 @@
 
 
 class IslandConfigOptions:
-    def __init__(self, config_contents: dict = None):
-        if not config_contents:
-            config_contents = {}
-        self.data_dir = config_contents.get(_DATA_DIR, DEFAULT_DATA_DIR)
+    def __init__(self, config_contents: dict = {}):
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/setup/island_config_options.py:22]: The core issue is that in Python, the value of a default argument is evaluated only once, when the function is declared. The default argument syntax config_contents = {} causes the Python interpreter to create an empty dictionary, but it’s **the same dictionary object every time the function is called**.

File: monkey/monkey_island/cc/setup/island_config_options.py [hunk 2]
```diff
@@ -20,22 +19,39 @@
...
+
+        self._initialize_defaults()
+        self.update(config_contents)
+
+    def _initialize_defaults(self):
+        self.data_dir = DEFAULT_DATA_DIR
+        self.log_level = DEFAULT_LOG_LEVEL
+        self.start_mongodb = DEFAULT_START_MONGO_DB
+        self.crt_path = DEFAULT_CRT_PATH
+        self.key_path = DEFAULT_KEY_PATH
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/setup/island_config_options.py:37]: Why not just set these in __init__?

File: monkey/monkey_island/cc/setup/island_config_options.py [hunk 3]
```diff
@@ -20,22 +19,39 @@
...
 
-        self.crt_path = config_contents.get(_SSL_CERT, DEFAULT_CERTIFICATE_PATHS).get(
-            _SSL_CERT_FILE, DEFAULT_CRT_PATH
+        self.crt_path = config_contents.get(_SSL_CERT, {_SSL_CERT_FILE: self.crt_path}).get(
+            _SSL_CERT_FILE, self.crt_path
         )
-        self.key_path = config_contents.get(_SSL_CERT, DEFAULT_CERTIFICATE_PATHS).get(
-            _SSL_CERT_KEY, DEFAULT_KEY_PATH
+        self.key_path = config_contents.get(_SSL_CERT, {_SSL_CERT_KEY: self.key_path}).get(
+            _SSL_CERT_KEY, self.key_path
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/setup/island_config_options.py:54]: This was the problem? paths not expanded and expected object instead of string?

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Couple of simple but important changes

File: monkey/monkey_island/cc/setup/island_config_options.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/setup/island_config_options.py:22]: Does that really matter here, though? This isn't going to be called more than once when the Island is started, AFAIR.

File: monkey/monkey_island/cc/setup/island_config_options.py [hunk 4]
```diff
@@ -20,22 +19,39 @@
...
-    def __init__(self, config_contents: dict = None):
-        if not config_contents:
-            config_contents = {}
-        self.data_dir = config_contents.get(_DATA_DIR, DEFAULT_DATA_DIR)
+    def __init__(self, config_contents: dict = {}):
+        self.data_dir = None
+        self.log_level = None
+        self.start_mongodb = None
+        self.crt_path = None
+        self.key_path = None
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/setup/island_config_options.py:27]: Why is this needed when `self._initialize_defaults()` is called right after this?

File: monkey/monkey_island/cc/setup/island_config_options.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/setup/island_config_options.py:22]: Its the same if a parameter isn't specified. But as long as the object isn't modified (it's not) it doesn't matter that it's the same object.

File: monkey/monkey_island/cc/setup/island_config_options.py (hunk 4 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/setup/island_config_options.py:27]: I tried this a few different ways. I didn't particularly like one version more than another. This was just the last way I tried it.

Members should be "defined" in the same place in every class so you don't need to go hunting around to figure out what members the class includes. I can just set the defaults in `_init_()`.

File: monkey/monkey_island/cc/setup/island_config_options.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/setup/island_config_options.py:37]: I can make that change. See https://github.com/guardicore/monkey/pull/1668/files#r781174189 for an explaination.

File: monkey/monkey_island/cc/setup/island_config_options.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/setup/island_config_options.py:54]: Yeah. This object is flat, whereas the config file is not. So using `self.__dict__.update()` won't work.

**Review by VakarisZ — APPROVED** [review]
